v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -29,12 +29,31 @@ func AddCrossResourceReferences(configuration *config.Configuration, idFactory a
 		AddCrossResourceReferencesStageID,
 		"Replace cross-resource references with genruntime.ResourceReference",
 		func(ctx context.Context, definitions astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, error) {
-			result := make(astmodel.TypeDefinitionSet)
+			typesWithARMIDs := make(astmodel.TypeDefinitionSet)
 
 			var crossResourceReferenceErrs []error
 
 			isCrossResourceReference := func(typeName astmodel.TypeName, prop *astmodel.PropertyDefinition) bool {
+				// First check if we know that this property is an ARMID already
 				isReference, err := configuration.ARMReference(typeName, prop.PropertyName())
+				if primitive, ok := astmodel.AsPrimitiveType(prop.PropertyType()); ok {
+					if primitive == astmodel.ARMIDType {
+						if err == nil && !isReference {
+							// We've overridden the ARM spec details
+							return false
+						}
+						// Mark configuration as unconsumed here to force an error since
+						// Swagger is overriding it so we don't need the annotation anymore
+						err = configuration.MarkARMReferenceUnconsumed(typeName, prop.PropertyName())
+						if err != nil && !config.IsNotConfiguredError(err) {
+							crossResourceReferenceErrs = append(
+								crossResourceReferenceErrs,
+								errors.Wrap(err, "failed to reset ARM reference flag"))
+						}
+						return true
```

Comment:
Instead of marking the configuration as unconsumed, maybe it would be better to return errors from the stage indicating that the configuration of `$armReference: true` is no longer required.

After all, the value _has_ been consumed - it's just been found to be redundant.

Reply:
Yeah that's also less code because we don't need to plumb the Unconsumed function through the config. Changed.